tests/unit/s2n_security_policy_cert_preferences_test.c

```diff
@@ -314,11 +315,19 @@ int main(int argc, char **argv)
     {
         DEFER_CLEANUP(struct s2n_cert_chain_and_key *cert = NULL, s2n_cert_chain_and_key_ptr_free);
         /* use a very insecure cert that would not be included in any reasonable cert preferences */
-        EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&cert, "rsae", "pkcs", "1024", "sha1"));
+        if (s2n_is_in_fips_mode() && !s2n_libcrypto_is_awslc()) {
+            /* OpenSSL FIPS module requires the key size for RSA to be bigger than or equal to 1024 bits.
```

Comment:
Is it just 1.0.2 FIPs that has this restriction, or does 3.x also have this restriction?

Reply:
Not entirely sure if 3.x has the same restriction, but I wasn't able to find the same function in 3.x. This check wouldn't include 3.x though, because the way we are detecting FIPS is not supported in 3.0 and higher. We call FIPS_mode() to determine if libcrypto is in fips mode. [This function is not present in 3.x](https://www.openssl.org/docs/man3.0/man7/fips_module.html#:~:text=Note%20that%20the%20old%20functions%20FIPS_mode()%20and%20FIPS_mode_set()%20are%20no%20longer%20present%20so%20you%20must%20remove%20them%20from%20your%20application%20if%20you%20use%20them.)

Reply:
In 3.X, it seems the minimum key size for RSA is 2048:
https://github.com/openssl/openssl/blob/4f5febe2c684a803553171940634c1b6f4b7ba40/include/openssl/rsa.h#L48